src/lavinmq/amqp/queue/stream_queue_message_store.cr

```diff
@@ -236,6 +247,14 @@ module LavinMQ::AMQP
         headers.not_nil!("Message lacks headers")["x-stream-offset"].as(Int64)
       end
 
+      private def filter_value_from_headers(headers) : String?
+        if filter = headers.try &.["x-stream-filter-value"]?
+          filter.to_s
+        else
+          nil
+        end
+      end
```

Comment:
```suggestion
      private def filter_value_from_headers(headers) : String?
        headers.try &.fetch("x-stream-filter-value", nil).try &.to_s
      end
```

Reply:
Or maybe this is more readable? 
```suggestion
      private def filter_value_from_headers(headers : Hash) : String?
        if value = headers["x-stream-filter-value"]?
          value.to_s
        end
      end
      
      private def filter_value_from_headers(headers : Nil) : Nil
      end
```

Reply:
I think the first suggestion is readable enough!